src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,94 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+
+import command.AddCommand;
+import command.Command;
+import command.DeleteCommand;
+import command.DoneCommand;
+import command.ExitCommand;
+import command.FindCommand;
+import command.ListCommand;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.Todo;
+
+
+/**
+ * Deals with making sense of the user command.
+ *
+ * @author felix-ong
+ */
+public class Parser {
+
+    /**
+     * Parses the user input to determine the command.
+     *
+     * @param userInput Command and description entered by the user.
+     * @return Command to be executed.
+     * @throws DukeException If missing arguments for commands or unknown commands.
+     */
+    public static Command parse(String userInput) throws DukeException {
```

Comment:
In my opinion, this method seems to be a little too long. If possible, maybe you can try to divide it into more methods.